tripal/tripal
# Revert one change from PR #1882

# Bug Fix

### Closes #1913  #

### Tripal Version: 4.x

## Description
This reverts a single change that causes the issue described in issue #1913 - You can't edit an entity and add a linked record.

~~I am submitting this as draft until testing reveals that there are no unintended side effects.~~

## Testing?
 1. Create an analysis entity
 2. Create a project entity. Do not include the analysis yet.
 3. Edit the project entity and add the analysis to it.
 4. The analysis field will not show up on the page before this PR, it will show up on this branch.
 5. Look in the Drupal field table, a record for it is missing (the bug before this PR) - or the record is present (intended behavior, only on this branch)
```
sitedb=> select * from tripal_entity__project_analysis;
 bundle  | deleted | entity_id | revision_id | langcode | delta | project_analysis_record_id | project_analysis_entity_id | project_analysi...
---------+---------+-----------+-------------+----------+-------+----------------------------+----------------------------+----------------...
 project |       0 |         1 |           1 | und      |     0 |                          1 |                          4 |                ...
(1 row)
```
 6. Look in the Chado `project_analysis` table and there is a record for it (intended behavior, happens before and after this branch)
```
 sitedb=> select * from project_analysis;
 project_analysis_id | project_id | analysis_id | rank 
---------------------+------------+-------------+------
                   1 |          1 |           1 |    1
(1 row)
```
 7. Let's test publishing too
```
INSERT INTO chado.project (name, description) VALUES ('Published Project', 'For Testing');
INSERT INTO chado.contact (name, description) VALUES ('Published Contact', 'Also For Testing');
INSERT INTO chado.project_contact (project_id, contact_id) VALUES ((SELECT project_id FROM chado.project WHERE name='Published Project'), (SELECT contact_id FROM chado.contact WHERE name='Published Contact'));
```
Then publish both project and contact and make sure the linked entity is present on the page. For example
![2024-06-30_PublishedProject](https://github.com/tripal/tripal/assets/8419404/d69d61f4-151d-4168-9d17-5ddff8d60444)
